tests/library/test_tasks.py

```diff
@@ -51,3 +53,51 @@ def test_multiple_choice_task(self):
 
         for output, target in zip(outputs, targets):
             self.assertDictEqual(output, target)
+
+    def test_task_metrics_type_checking(self):
+        operator = FormTask(
+            inputs={"input": str},
+            outputs={"label": str},
+            prediction_type="str",
+            metrics=["metrics.wer", "metrics.rouge", "metrics.roc_auc"],
+        )
+
+        operator.check_metrics_type()
+
+        operator.prediction_type = "Dict"
+        with self.assertRaises(ValueError):
+            operator.check_metrics_type()
```

Comment:
Best to also check the message - as in:

 with self.assertRaises(RuntimeError) as e:
            check_operator(
                operator=FilterByCondition(
                    values={"b": ["weird_value"]}, condition="in"
                ),
                inputs=inputs,
                targets=[],
                tester=self,
            )
        self.assertEqual(
            str(e.exception),
            "FilterByCondition filtered out every instance in stream 'test'. If this is intended set error_on_filtered_all=False",
        )